api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -76,18 +169,31 @@
   }
 
   @Override
-  public ResponseEntity<CreateAccountResponse> createAccount(CreateAccountRequest request) {
+  public ResponseEntity<Profile> createAccount(CreateAccountRequest request) {
     if (request.getInvitationKey() == null
         || !request.getInvitationKey().equals(cloudStorageService.readInvitationKey())) {
-      return ResponseEntity.badRequest()
-          .body(new CreateAccountResponse().message(
-              "Missing or incorrect invitationKey (this API is not yet publicly launched)"));
+      throw new BadRequestException(
+          "Missing or incorrect invitationKey (this API is not yet publicly launched)");
     }
-    directoryService.createUser(
+    com.google.api.services.admin.directory.model.User googleUser = directoryService.createUser(
         request.getGivenName(), request.getFamilyName(), request.getUsername(),
         request.getPassword()
     );
-    return ResponseEntity.status(HttpStatus.CREATED).build();
+
+    // Create a user that has no data access or FC user associated.
```

Comment:
I would prefer to not create the user account here. I'd prefer we keep this endpoint focused on creating just the Google account. Instead, I would suggest waiting until the user first logs in and looking up their account in our database. If it doesn't exist, we can create it at that time.

Reply:
That would prevent us from easily figuring out what users have created accounts but never signed in. (Which would prevent us from cleaning up those accounts.)

I think we have to record something here.